Sources/Networking/HTTPClient/HTTPStatusCode.swift

```diff
@@ -93,3 +93,13 @@ extension HTTPStatusCode {
     }
 
 }
+
+extension URLResponse {
+
+    var httpStatusCode: HTTPStatusCode? {
+        guard let response = self as? HTTPURLResponse else { return nil }
+
+        return .init(rawValue: response.statusCode)
```

Comment:
[nitpick, FFTI] it might just be me, but I find this pattern in general to be less readable than 

```swift
    return HTTPStatusCode(rawValue: response.statusCode)
```

since I have to do a double-take and check what the return type of the method is to figure out what we're creating

Reply:
Yeah it's definitely a tradeoff, especially for more complex functions where the context of the type might be lost to the reader. In this particular case my thinking is just "let the compiler figure it out".